File: filter-pattern/python/src/CriteriaFemale.py
# -*- coding: utf-8 -*-
"""
@author: jarry
"""

from src.Criteria import Criteria


# 根据标准接口实现的过滤
class CriteriaFemale(Criteria):

    def filter(self, persons):
        female_persons = []
        for person in persons:
            if person.get_gender().upper() == 'FEMALE':
                female_persons.append(person)

        return female_persons

File: filter-pattern/go/src/CriteriaFemale.go
package src

import "strings"

// 根据标准接口实现的过滤
type CriteriaFemale struct {
}

// 根据是否女性进行过滤
func (r *CriteriaFemale) Filter(persons []*Person) []*Person {
  var femalePersons []*Person
  for _, person := range persons {
    if strings.ToUpper(person.GetGender()) == "FEMALE" {
      femalePersons = append(femalePersons, person)
    }
  }
  return femalePersons
}
